setup.py

```diff
@@ -1,20 +1,29 @@
 #!/usr/bin/python3
 # encoding: utf-8
+"""Setup file for dulwich.
+
+Copyright (C) 2008-2016 Jelmer Vernooĳ <jelmer@jelmer.uk>
+"""
+
 # Setup file for dulwich
 # Copyright (C) 2008-2016 Jelmer Vernooĳ <jelmer@jelmer.uk>
 
+import io
+import os
+import sys
+
 try:
-    from setuptools import setup, Extension
+    from setuptools import Extension
+    from setuptools import setup
 except ImportError:
-    from distutils.core import setup, Extension
-    has_setuptools = False
+    from distutils.core import Extension
+    from distutils.core import setup
+    HAS_SETUPTOOLS = False
 else:
-    has_setuptools = True
+    HAS_SETUPTOOLS = True
```

Comment:
This is not a constant, it's a variable that's set based on whether setuptools is present.

Reply:
```python
try:
    from setuptools import Extension
    from setuptools import setup
except ImportError:
    from distutils.core import Extension
    from distutils.core import setup
    HAS_SETUPTOOLS = False
else:
    HAS_SETUPTOOLS = True
from distutils.core import Distribution
from typing import Any
from typing import Dict
```
Error messenger: `C0103: Constant name "dulwich_version_string" doesn't conform to UPPER_CASE naming style (invalid-name)`